### PR TITLE
Add `udx.networkInterfaces()`

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

File: binding.c [hunk 1]
```diff
@@ -514,21 +514,21 @@ NAPI_METHOD(udx_napi_network_interfaces) {
   napi_create_array(env, &result);
 
   while (i < count) {
-    uv_interface_address_t interface = interfaces[i++];
```
Comment (kasperisager) [inline binding.c]: I don't even know what to think... https://stackoverflow.com/a/25234279/12524929

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

File: binding.c [hunk 2]
```diff
@@ -37,7 +37,7 @@
   }
 
 typedef struct {
-  udx_socket_t udx;
+  udx_socket_t socket;
```
Comment (kasperisager) [inline binding.c:40]: Unrelated rename.

File: binding.c (hunk 1 above)
Comment (mafintosh) [reply binding.c]: ah yes - naming paranoia. hilarious
